### PR TITLE
Network: OVN load balancers

### PR DESCRIPTION
**Pull request by tomponline**
Specification: https://discuss.linuxcontainers.org/t/lxd-network-load-balancers-ovn/14317

- [x] Implementation
- [x] Tests (https://github.com/lxc/lxc-ci/pull/566)
- [x] Docs

### CODE REVIEW

File: shared/api/network_load_balancer.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	"net"
+	"strings"
+)
+
+// NetworkLoadBalancerBackendName represents a target backend name in a network load balancer
+//
+// swagger:model
+//
+// API extension: network_load_balancer.
+type NetworkLoadBalancerBackendName string
```
Comment (stgraber) [inline shared/api/network_load_balancer.go:13]: What's the purpose for this custom type?

It's not one of those where we're dealing with a fixed set of options/enums and there are no functions attached to it, so it looks like it's just going to make our users' life potentially harder by having to cast between NetworkLoadBalancerBackendName and regular strings.

File: lxd/network/driver_common.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"os"
```
Comment (stgraber) [inline lxd/network/driver_common.go:7]: Typo in commit title string

File: lxd/network_load_balancers.go [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+		u := api.NewURL().Path(version.APIVersion, "networks", n.Name(), "load-balancers", listenAddress)
+		loadBalancerURLs = append(loadBalancerURLs, u.String())
+	}
+
+	return response.SyncResponse(true, loadBalancerURLs)
+}
+
+// swagger:operation POST /1.0/networks/{networkName}/load-balancers network-load-balancers network_load_balancers_post
+//
+// Add a network address load balancer
```
Comment (stgraber) [inline lxd/network_load_balancers.go:184]: I'm not sure about the phrasing of `network address load balancer` as used in this file, can't it just be `network load balancer`?

As we handle traffic per port, `network address` feels odd.

File: lxc/network_load_balancer.go [hunk 4]
```diff
@@ -317,7 +317,7 @@ func (c *cmdNetworkLoadBalancerCreate) Run(cmd *cobra.Command, args []string) er
 	return nil
 }
 
-// Get
+// Get.
```
Comment (stgraber) [inline lxc/network_load_balancer.go:320]: Shouldn't be in this commit

File: shared/api/network_load_balancer.go (hunk 1 above)
Comment (tomponline) [reply shared/api/network_load_balancer.go:13]: I've removed it.

File: lxd/network/driver_common.go (hunk 2 above)
Comment (tomponline) [reply lxd/network/driver_common.go:7]: Fixed.

**Comment by tomponline** [comment]
@stgraber all comments addressed. Thanks

**Pull request merged**